src/main/java/io/usethesource/vallang/type/TypeFactory.java

```diff
@@ -54,7 +55,7 @@
 	/**
 	 * Caches all types to implement canonicalization
 	 */
-	private final Map<Type, Type> fCache = new HashMap<Type, Type>();
+	private final WeakReferenceCache<Type, Type> fCache = new WeakReferenceCache<Type, Type>(true, true);
```

Comment:
initial size is now `32`, perhaps push it up to what is typically needed to start the Rascal interpreter and generate one parser.

Reply:
Good plan, I'll measure that, and pass it in as an argument.